src/main/java/org/jabref/logic/importer/util/GrobidService.java

```diff
@@ -0,0 +1,47 @@
+package org.jabref.logic.importer.util;
+
+import java.io.IOException;
+
+import org.jabref.logic.net.URLDownload;
+import org.jabref.preferences.JabRefPreferences;
+
+/**
+ * Implements an API to a GROBID server, as described at
+ * https://grobid.readthedocs.io/en/latest/Grobid-service/#grobid-web-services
+ *
+ * Note: Currently a custom GROBID server is used...
+ * https://github.com/NikodemKch/grobid
+ *
+ * The methods are structured to match the GROBID server api.
+ * Each method corresponds to a GROBID service request. Only the ones already used are already implemented.
+ */
+public class GrobidService {
+
+    private static JabRefPreferences jabRefPreferences;
+
+    public GrobidService(JabRefPreferences jabRefPreferences) {
+        GrobidService.jabRefPreferences = jabRefPreferences;
+    }
+
+    public String processCitation(String rawCitation, int consolidateCitations) throws GrobidServiceException {
```

Comment:
I guess `consolidateCitations` is actually a boolean, or an enum? Then this should be the interface of this method and internally converted to the correct integer before posting to grobid.

Reply:
58eb443a1b60784567ec50d8454d6b2d0bc6b876